site/topics/sorting/sorting.rst

```diff
@@ -177,56 +177,128 @@ Insertion Sort
 Bubble Sort
 ===========
 
-	.. image:: bubble.gif
+.. image:: bubble_sort.gif
+   :height: 333 px
+   :align: center
+   :target: https://en.wikipedia.org/wiki/Bubble_sort
 
-* Maybe you find Insertion sort or Selection sort ugly or offensive?
-* No problem. Remember: there are *many* algorithms to solve any one problem.
-* You give me a list called ``in_list``
-* I scan through the list, looking at adjacent pairs of values.
-* If I see a pair that is "out of order" (e.g., ``[17, 9]`` ), I swap the two values to be in order ( ``[9,17]`` ).
-* I keep doing that until the list is sorted.
+* Bubble sort works a little differently than selection or insertion sort
+* The general idea is to perform multiple scans of the unsorted list comparing each adjacent pair of elements
 
-.. admonition:: Activity
-    :class: activity
+    * If the elements are out of order, swap them, otherwise, leave them alone
+    * Move to the next adjacent pair of elements
+    * Repeat
 
-    Do a bubble sort, with pencil and paper, on the list ``[3,7,15,9,4,11,1,5,2]``. Record the value of your list at each step.  
+* Consider the below unsorted list
 
-* It's called "bubble sort" because the smaller values seem to "bubble up to the top".
-* Kinda cool because:
-    * We end up effecting a *global* change on the list (it goes from unsorted to sorted)...
-    * ... but we only use *local* information about the elements (we only ever compare neighbours in the list)
-   
-Let's see Bubble sort in Python::
-
-    def bubble_sort(in_list):
-        swapped_something = True
-        while swapped_something:
-            swapped_something = False
-         
-            for i in range(len(in_list)-1):
-                if in_list[i] > in_list[i+1]:
-                    tmp = in_list[i]
-                    in_list[i]=in_list[i+1]
-                    in_list[i+1]=tmp
-                    swapped_something = True
-        return in_list
-
-* Ugh... Wouldn't the above code be better if there were comments?
+    :math:`4, 3, 2, 1`
+
+    * Four is greater than three, so they swap
+
+    :math:`3, 4, 2, 1`
+
+    * Four is greater than two, so they swap
+
+    :math:`3, 2, 4, 1`
+
+    * Four is greater than one, so they swap
+
+    :math:`3, 2, 1, 4`
+
+* There are two important things to notice at the end of the first pass
+
+    * A single pass is not enough to guarantee the list is sorted
+    * After the first pass, the largest element in the unsorted list will be in its correct location
```

Comment:
Is this intuitive enough? Or does it need more explanation. I am hoping it's intuitive enough as I am too tired to explain more and I have two children screaming at me.